### PR TITLE
Remove usages of `std::call_once`.

### PR DESCRIPTION
**Pull request by mzient**
## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)
Workaround for a library bug.

## Additional information:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78184

`std::call_once` has a bug in some versions of glibc. A custom implementation has been provided.
Usages with static once_flag have been replaced with "magic statics". Optical flow doesn't need atomicity and the once semantics were not required at all.
Other usages use the custom implementation.

A stale warning about antialiasing (also guarded with call once) was removed.

### Tests:

- [ ] Existing tests apply
- [X] New tests added
  - [ ] Python tests
  - [X] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [X] Documentation updated
  - [ ] Docstring
  - [X] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/image/resize/resampling_attr.cc [hunk 1]
```diff
@@ -93,16 +94,6 @@ void ResamplingFilterAttr::PrepareFilterParams(
...
-    if ((has_min && is_linear_default(make_cspan(min_arg_))) ||
-        (has_interp && is_linear_default(make_cspan(interp_type_arg_)))) {
-      static std::once_flag linear_default_warning_flag;
-      std::call_once(linear_default_warning_flag, [&]() {
-        DALI_WARN(
-            "The default behavior for LINEAR interpolation type has been changed to apply an "
-            "antialiasing filter. If you didn't mean to apply an antialiasing filter, please use "
-            "`antialias=False`");
-      });
-    }
```
Comment (mzient) [inline dali/operators/image/resize/resampling_attr.cc:105]: This was lamentably outdated.

File: dali/operators/sequence/optical_flow/optical_flow.h [hunk 2]
```diff
@@ -145,18 +145,11 @@ class OpticalFlow : public StatelessOperator<Backend> {
...
-                       optical_flow_.reset(
-                               new optical_flow::OpticalFlowImpl(of_params_,
-                                                                 width,
-                                                                 height,
-                                                                 channels,
-                                                                 image_type,
-                                                                 device_id,
-                                                                 stream));
-                        optical_flow_->Init(of_params_);
-                   });
```
Comment (mzient) [inline dali/operators/sequence/optical_flow/optical_flow.h:159]: This is called from within setup. No atomicity is required, setup is never called in parallel with itself.

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/image/resize/resampling_attr.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/image/resize/resampling_attr.cc:105]: I have mixed filling about including this in this particular PR. But up to you.

**Comment by JanuszL** [comment]
Please excuse my naïve question, as I understand statics are used when the code doesn't throw, call_once where it might, rigth?

**Comment by mzient** [comment]
> Please excuse my naïve question, as I understand statics are used when the code doesn't throw, call_once where it might, right?

I use "magic statics" when the call_once flag was also static. The compiler should take care of everything. I can double check it.

**Review by JanuszL — APPROVED** [review]

**Comment by mzient** [comment]
Checked: magic statics do the trick on `manylinux`, whereas call_once hangs in a simple test program.

**Pull request merged**